File: setup-logos.py
#!/usr/bin/env python3
"""
Setup script to create B&W and color versions of the logo
Run this once: python setup-logos.py
"""

import os
import sys
from PIL import Image
import shutil

def setup_logos():
    public_dir = os.path.join(os.path.dirname(__file__), 'public')
    logo_path = os.path.join(public_dir, 'logo.png')
    logo_color_path = os.path.join(public_dir, 'logo-color.png')
    logo_bw_path = os.path.join(public_dir, 'logo-bw.png')

    print('Setting up logos...')

    # Check for either logo.png or logo-color.png as source
    source_logo = None
    if os.path.exists(logo_path):
        source_logo = logo_path
    elif os.path.exists(logo_color_path):
        source_logo = logo_color_path
    else:
        print('ERROR: Neither logo.png nor logo-color.png found in public folder')
        sys.exit(1)

    try:
        # Create color version if it doesn't exist
        if not os.path.exists(logo_color_path) and source_logo == logo_path:
            shutil.copy(logo_path, logo_color_path)
            print('Created logo-color.png')

        # Create B&W version
        if os.path.exists(logo_bw_path):
            os.remove(logo_bw_path)

        img = Image.open(source_logo)
        img_bw = img.convert('L')
        img_bw.save(logo_bw_path)
        print('Created logo-bw.png')
        print('SUCCESS: Logos setup complete!')

    except Exception as err:
        print(f'ERROR: {err}')
        sys.exit(1)

if __name__ == '__main__':
    setup_logos()

File: setup-logos.js
#!/usr/bin/env node

import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';
import { execSync } from 'child_process';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

async function setupLogos() {
  const publicDir = path.join(__dirname, 'public');
  const logoPath = path.join(publicDir, 'logo.png');
  const logoBwPath = path.join(publicDir, 'logo-bw.png');
  const logoColorPath = path.join(publicDir, 'logo-color.png');

  console.log('Setting up logos...');

  if (!fs.existsSync(logoPath)) {
    console.error('ERROR: logo.png not found in public folder');
    process.exit(1);
  }

  try {
    const sharp = (await import('sharp')).default;

    fs.copyFileSync(logoPath, logoColorPath);
    console.log('Created logo-color.png');

    await sharp(logoPath)
      .grayscale()
      .toFile(logoBwPath);
    console.log('Created logo-bw.png');
    console.log('SUCCESS: Logos setup complete!');

  } catch (err) {
    if (err.message.includes('Cannot find module') || err.message.includes('sharp')) {
      console.log('Installing sharp...');
      try {
        execSync('npm install --save-dev sharp', { stdio: 'inherit' });
        const sharp = (await import('sharp')).default;
        fs.copyFileSync(logoPath, logoColorPath);
        await sharp(logoPath).grayscale().toFile(logoBwPath);
        console.log('SUCCESS: Logos created!');
      } catch (installErr) {
        console.error('ERROR: Failed -', installErr.message);
        process.exit(1);
      }
    } else {
      console.error('ERROR:', err.message);
      process.exit(1);
    }
  }
}

setupLogos();
